### ISSUE TITLE
[Impeller] Flutter GPU: Add `flutter_gpu` package.

### ISSUE THREAD

**Issue by bdero** [flutter/flutter#131346]
Part of http://flutter.dev/go/impeller-dart
Umbrella bug #130921.

~Add the `dart:gpu` package to the Flutter Engine Dart SDK build and patch the Dart compiler to allow imports. Then, migrate in the existing API surface.~

Add the new `flutter_gpu` package. Then, migrate the existing API surface.

**Comment by bdero** [comment]
We talked with the Dart team about this approach and agreed to instead try shipping it as a regular package. I was able to get the symbols exported from the engine and verify that all of the same FFI tricks we use for `dart:ui` still work fine for this.

@jonahwilliams started a discussion around approaches for shipping the library alongside the engine here: https://github.com/flutter/flutter/issues/131711

In the meantime, I'm going to focus on getting a test harness setup on the Engine side. Should be quite similar to `dart:ui`, except we can just import the library via relative path instead of by package.
